File: flutter_hr/lib/models/salary.dart
class Salary {
  final int? id;
  final int employeeId;
  final int month;
  final int year;
  final double baseSalary;
  final double advanceSalary;
  final double bonusAmount;
  final double allowances;
  final double deductions;
  final double tax;
  final double netSalary;
  final String status;
  final String? paymentMethod;
  final String? paymentDate;

  Salary({
    this.id,
    required this.employeeId,
    required this.month,
    required this.year,
    required this.baseSalary,
    this.advanceSalary = 0,
    this.bonusAmount = 0,
    this.allowances = 0,
    this.deductions = 0,
    this.tax = 0,
    required this.netSalary,
    required this.status,
    this.paymentMethod,
    this.paymentDate,
  });

  factory Salary.fromJson(Map<String, dynamic> json) {
    return Salary(
      id: json['id'] as int?,
      employeeId: json['employeeId'] as int? ?? 0,
      month: json['month'] as int? ?? 0,
      year: json['year'] as int? ?? 0,
      baseSalary: (json['baseSalary'] as num?)?.toDouble() ?? 0.0,
      advanceSalary: (json['advanceSalary'] as num?)?.toDouble() ?? 0.0,
      bonusAmount: (json['bonusAmount'] as num?)?.toDouble() ?? 0.0,
      allowances: (json['allowances'] as num?)?.toDouble() ?? 0.0,
      deductions: (json['deductions'] as num?)?.toDouble() ?? 0.0,
      tax: (json['tax'] as num?)?.toDouble() ?? 0.0,
      netSalary: (json['netSalary'] as num?)?.toDouble() ?? 0.0,
      status: json['status']?.toString() ?? 'Draft',
      paymentMethod: json['paymentMethod']?.toString(),
      paymentDate: json['paymentDate']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'employeeId': employeeId,
      'month': month,
      'year': year,
      'baseSalary': baseSalary,
      'advanceSalary': advanceSalary,
      'bonusAmount': bonusAmount,
      'allowances': allowances,
      'deductions': deductions,
      'tax': tax,
      'netSalary': netSalary,
      'status': status,
      'paymentMethod': paymentMethod,
      'paymentDate': paymentDate,
    };
  }
}

File: flutter_hr/HR-And-Payroll-Management-System-Backend/src/main/java/Israt/example/Hr/and/payroll/management/system/backend/entity/Salary.java
package Israt.example.Hr.and.payroll.management.system.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "salaries")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Salary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_id", nullable = false)
    private Long employeeId;

    @Column(nullable = false)
    private Integer month;

    @Column(nullable = false)
    private Integer year;

    @Column(name = "base_salary", nullable = false)
    private BigDecimal baseSalary = BigDecimal.ZERO;

    @Column(name = "advance_salary")
    private BigDecimal advanceSalary = BigDecimal.ZERO;

    @Column(name = "bonus_amount")
    private BigDecimal bonusAmount = BigDecimal.ZERO;

    @Column
    private BigDecimal allowances = BigDecimal.ZERO;

    @Column
    private BigDecimal deductions = BigDecimal.ZERO;

    @Column
    private BigDecimal insurance = BigDecimal.ZERO;

    @Column
    private BigDecimal medicare = BigDecimal.ZERO;

    @Column
    private BigDecimal tax = BigDecimal.ZERO;

    @Column(name = "provident_fund")
    private BigDecimal providentFund = BigDecimal.ZERO;

    @Column(name = "food_deduction")
    private BigDecimal foodDeduction = BigDecimal.ZERO;

    @Column(name = "other_deduction")
    private BigDecimal otherDeduction = BigDecimal.ZERO;

    @Column(name = "overtime_hours")
    private BigDecimal overtimeHours = BigDecimal.ZERO;

    @Column(name = "overtime_rate")
    private BigDecimal overtimeRate = BigDecimal.ZERO;

    @Column(name = "overtime_pay")
    private BigDecimal overtimePay = BigDecimal.ZERO;

    @Column(name = "gross_salary")
    private BigDecimal grossSalary = BigDecimal.ZERO;

    @Column(name = "net_salary")
    private BigDecimal netSalary = BigDecimal.ZERO;

    @Enumerated(EnumType.STRING)
    @Column
    private SalaryStatus status = SalaryStatus.Draft;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_method")
    private PaymentMethod paymentMethod;

    @Column(name = "payment_reference")
    private String paymentReference;

    @Column(name = "payment_date")
    private LocalDate paymentDate;

    @Column
    private String notes;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "approved_by")
    private Long approvedBy;

    @Column(name = "approved_date")
    private LocalDate approvedDate;

    public enum SalaryStatus {
        Draft, Pending, Processing, Paid, Failed
    }

    public enum PaymentMethod {
        Bank_Transfer, Check, Cash
    }
}